tsup.config.ts

```diff
@@ -0,0 +1,26 @@
+import { defineConfig } from 'tsup';
+
+export default defineConfig({
+  entry: ['src/*'],
+  bundle: false,
+  clean: true,
+  format: ['cjs', 'esm'],
+  dts: true,
+  tsconfig: './tsconfig.json',
+  plugins: [
+    {
+      // https://github.com/egoist/tsup/issues/953#issuecomment-2132576167
+      // ensuring that all local requires in `.cjs` files import from `.cjs` files.
+      // require('./path') → require('./path.cjs') in `.cjs` files
+      name: 'fix-cjs-require',
```

Comment:
I'm surprised that u need any kind of fixup here. What would break if this was removed?

Reply:
We have a package that uses `bind-event-listener` in a plugin for `eslint@8` which was failing without this fix 😢

Reply:
Ah, I see. I'd probably stay away from `type: module` here since you are creating a dual package anyway. That would fix your issue but it would introduce the mirror issue in `.mjs` files (it just perhaps would be less surprising there that you need to add extensions in your imports)

Reply:
Dang. I was keen to make cjs more of an after thought which I could drop. Maybe easiest to go the other way 😭

Reply:
I mean, since u already have those workarounds in place - it's probably fine the way it is. I only have a vague recollection of some React Native versions having problems with non `.js` files in `main` but perhaps u can ignore it. Similarly, IIRC, old CRAs were loading `.cjs` as files and not javascript sources (as in, they were replacing them with URLs to the generated assets, like they do with `.png` files and more)

Reply:
Oh dear. I'm also tempted to cut my losses and just got full ESM 🔥

Reply:
But I'm leaning towards keeping it as is

Reply:
I am thinking of leaving it is as "it is an esm module, but it has some cjs fallbacks when we need it". If it doesn't work out, we can go full esm. 

Aside: publishing packages that "just work" is a nightmare right now 😭

Reply:
Any objections @Andarist to merging in it's current state? My open thought is whether I should update `"target"` to be "ES6" 🤔

Reply:
I’d bump it even higher although this lib is so small that it probably doesnt use any es2016+ features anyway ;p

Reply:
Right now I have target set at es2022